source/frontend/assets/scripts/database.js

```diff
@@ -116,16 +116,16 @@ async function deleteRecipe(recipeJSON) {
  */
 async function completeRecipe(recipeJSON) {
   try {
-    const response = await fetch(`${url}/users/completedRecipes/${recipeJSON.recipeId}`, {
+    if (localStorage.getItem('userToken') == null) {
+      return false;
+    }
+    await fetch(`${url}/users/completedRecipes/${recipeJSON.recipeId}`, {
```

Comment:
Your current implementation here ignores whatever result is from fetch, and just simply returns true. However, if you look at the route you are making request to here: It has multiple responses ex:
- 409: for already completed recipe
- 200: for successfully completed recipe
- 500: for server error.

By deleting the if statement in the original program, this part of the code ignores situations for 409 and 500 and display success in the frontend which is going to cause big confusions.